pkg/flavor/kubernetes/nfs.go

```diff
@@ -117,51 +130,109 @@ func (flavor *Flavor) CreateNFSVolume(pvName string, reqVolSize int64, parameter
 		}
 	}
 
-	// Return newly created underlying nfs claim uid with pv attributes
+	// decorate NFS PV with its volume handle as label for easy lookup during RWX PV deletion
+	pv.ObjectMeta.Labels = make(map[string]string)
+	pv.ObjectMeta.Labels[nfsVolumeHandleLabelKey] = pv.Spec.PersistentVolumeSource.CSI.VolumeHandle
+	flavor.kubeClient.CoreV1().PersistentVolumes().Update(pv)
+
+	// Return newly created underlying nfs pv handle with pv attributes
 	return &csi.Volume{
-		VolumeId:      fmt.Sprintf("%s", claim.ObjectMeta.UID),
+		VolumeId:      pv.Spec.PersistentVolumeSource.CSI.VolumeHandle,
 		CapacityBytes: reqVolSize,
 		VolumeContext: volumeContext,
+		ContentSource: volumeContentSource,
 	}, false, nil
 }
 
 // DeleteNFSVolume deletes nfs volume which represents nfs pvc, deployment and service
-func (flavor *Flavor) DeleteNFSVolume(pvName string) error {
-	log.Tracef(">>>>> DeleteNFSVolume with %s", pvName)
+func (flavor *Flavor) DeleteNFSVolume(volumeID string) error {
+	log.Tracef(">>>>> DeleteNFSVolume with %s", volumeID)
 	defer log.Tracef("<<<<< DeleteNFSVolume")
 
+	resourceName, err := flavor.getNFSResourceNameByVolumeID(volumeID)
+	if err != nil {
+		return err
+	}
+
+	nfsNamespace, err := flavor.getNFSNamespaceByVolumeID(volumeID)
+	if err != nil {
+		return err
+	}
+
 	// delete deployment deployment/hpe-nfs-<originalclaim-uid>
-	resourceName := fmt.Sprintf("%s%s", nfsPrefix, strings.TrimPrefix(pvName, "pvc-"))
-	err := flavor.DeleteNFSDeployment(resourceName)
+	err = flavor.DeleteNFSDeployment(resourceName, nfsNamespace)
 	if err != nil {
 		log.Errorf("unable to delete nfs deployment %s as part of cleanup, err %s", resourceName, err.Error())
 	}
 
 	// delete nfs pvc pvc/hpe-nfs-<originalclaim-uuid>
 	// if deployment is still around, then pvc cannot be deleted due to protection, try to cleanup as best effort
-	err = flavor.DeleteNFSPVC(resourceName)
+	err = flavor.DeleteNFSPVC(resourceName, nfsNamespace)
 	if err != nil {
 		log.Errorf("unable to delete nfs pvc %s as part of cleanup, err %s", resourceName, err.Error())
 	}
 
 	// delete service service/hpe-nfs-<originalclaim-uid>
-	err = flavor.DeleteNFSService(resourceName)
+	err = flavor.DeleteNFSService(resourceName, nfsNamespace)
 	if err != nil {
 		log.Errorf("unable to delete nfs service %s as part of cleanup, err %s", resourceName, err.Error())
 	}
 
 	return err
 }
 
+func (flavor *Flavor) getNFSResourceNameByVolumeID(volumeID string) (string, error) {
+	// get NFS PV by volume-id
+	pv, err := flavor.getPvByNFSLabel(nfsVolumeHandleLabelKey, volumeID)
```

Comment:
nit:  Why is NFS all caps but PV is not?

Reply:
my bad..will update..:)